packages/visx-xychart/src/components/series/BaseBarSeries.tsx

```diff
@@ -0,0 +1,129 @@
+  const getScaledY = useCallback(getScaledValueFactory(yScale, yAccessor), [yScale, yAccessor]);
+  const [xMin, xMax] = xScale.range().map(Number);
+  const [yMax, yMin] = yScale.range().map(Number);
+
+  const scaleBandwidth = getScaleBandwidth(horizontal ? yScale : xScale);
+  const barThickness =
+    scaleBandwidth ||
+    getFallbackBandwidth((horizontal ? innerHeight : innerWidth) / data.length, barPadding);
+
+  // try to figure out the 0 baseline for correct rendering of negative values
+  // we aren't sure if these are numeric scales or not ahead of time
+  const maybeXZero = xScale(0);
+  const maybeYZero = yScale(0);
+  const xZeroPosition = isValidNumber(maybeXZero)
+    ? // if maybeXZero _is_ a number, but the scale is not clamped and it's outside the domain
+      // fallback to the scale's minimum
+      Math.max(maybeXZero, Math.min(xMin, xMax))
+    : Math.min(xMin, xMax);
+  const yZeroPosition = isValidNumber(maybeYZero)
+    ? Math.min(maybeYZero, Math.max(yMin, yMax))
+    : Math.max(yMin, yMax);
+
+  const color = colorScale?.(dataKey) ?? theme?.colors?.[0] ?? '#222';
+
+  const bars = useMemo(() => {
+    const xOffset = horizontal ? 0 : -barThickness / 2;
+    const yOffset = horizontal ? -barThickness / 2 : 0;
+    return data.map((datum, index) => {
+      const x = getScaledX(datum) + xOffset;
+      const y = getScaledY(datum) + yOffset;
+      const barLength = horizontal ? x - xZeroPosition : y - yZeroPosition;
+
+      return {
+        key: `${index}`,
+        x: horizontal ? xZeroPosition + Math.min(0, barLength) : x,
+        y: horizontal ? y : yZeroPosition + Math.min(0, barLength),
+        width: horizontal ? Math.abs(barLength) : barThickness,
+        height: horizontal ? barThickness : Math.abs(barLength),
+        fill: color, // @TODO allow prop overriding
+      };
+    });
+  }, [barThickness, color, data, getScaledX, getScaledY, horizontal, xZeroPosition, yZeroPosition]);
+
+  const { showTooltip, hideTooltip } = useContext(TooltipContext) ?? {};
+  const handleMouseMove = useCallback(
+    (params?: HandlerParams) => {
+      const { svgPoint } = params || {};
+      if (svgPoint && width && height && showTooltip) {
+        const datum = (horizontal ? findNearestDatumY : findNearestDatumX)({
+          point: svgPoint,
+          data,
+          xScale,
+          yScale,
+          xAccessor,
+          yAccessor,
+          width,
+          height,
+        });
+        if (datum) {
+          showTooltip({
+            key: dataKey,
+            ...datum,
+            svgPoint,
+          });
+        }
+      }
+    },
+    [dataKey, data, horizontal, xScale, yScale, xAccessor, yAccessor, width, height, showTooltip],
+  );
+  useEventEmitter('mousemove', handleMouseMove);
+  useEventEmitter('mouseout', hideTooltip);
+
+  return (
+    <g className="vx-bar-series">
+      <BarsComponent bars={bars} horizontal={horizontal} xScale={xScale} yScale={yScale} />
```

Comment:
this is all the same as previous `BarStack`, `bars` are just passed to `BarsComponent` now.